aiosmtpd/smtp.py

```diff
@@ -505,29 +503,34 @@ def smtp_DATA(self, arg):
             return
         yield from self.push('354 End data with <CR><LF>.<CR><LF>')
         data = []
-        self.num_bytes = 0
+        num_bytes = 0
+        size_exceeded = False
         while not self.connection_closed:
             line = yield from self._reader.readline()
             if line == b'.\r\n':
+                if data:
+                    data[-1] = data[-1].rstrip(b'\r\n')
                 break
-            self.num_bytes += len(line)
-            if self.data_size_limit and self.num_bytes > self.data_size_limit:
+            num_bytes += len(line)
+            if (not size_exceeded) and self.data_size_limit and \
+                    num_bytes > self.data_size_limit:
```

Comment:
I think it must be integer and also always positive integer for security reasons. Also it maybe good idea to fall it to reader entity to be sure its playing together.
On my local implementation I added exception in init if this value is not set explicitly, to be sure it will not be missed by admins during service configuration. May be it is too strong (we have strong security requirements), and it can be set to default value in common implementation, but Im aware of ability to disable it and create big security hole.